### PR TITLE
Add support for resource detector option.

### LINKED ISSUE #69: Add option to detect resources.
Java oc-agent exporter detects resources automatically using [ResourceUtil.detectResources()](https://static.javadoc.io/io.opencensus/opencensus-contrib-monitored-resource-util/0.20.0/io/opencensus/contrib/monitoredresource/util/MonitoredResourceUtils.html#detectResource--). However, in go it only [detects](https://github.com/census-ecosystem/opencensus-go-exporter-ocagent/blob/88230df77e734507cd633004f098a4febd80ba0b/ocagent.go#L511) resources from the environment. While it is desired to run all auto detect from every cloud provider (due to dependency), it is still desirable to detect resource using a specific detector or a custome detector. Hence this request to add the support.

### PR DESCRIPTION
**Pull request by rghetia**
fixes #69

### CODE REVIEW

File: resource_detector_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Copyright 2018, OpenCensus Authors
```
Comment (rghetia) [inline resource_detector_test.go]: s/2018/2019

File: ocagent.go [hunk 2]
```diff
@@ -125,7 +126,17 @@ func NewUnstartedExporter(opts ...ExporterOption) (*Exporter, error) {
 	viewDataBundler.BundleCountThreshold = 500 // TODO: (@odeke-em) make this configurable.
 	e.viewDataBundler = viewDataBundler
 	e.nodeInfo = NodeWithStartTime(e.serviceName)
-	e.resource = resourceProtoFromEnv()
+	if e.resourceDetector != nil {
+		res, err := e.resourceDetector(context.Background())
+		if err != nil {
+			panic(fmt.Sprintf("Error detecting resource. err:%v\n", err))
```
Comment (songy23) [inline ocagent.go]: Do we want to panic here? Or is it OK to just log a message and continue?

Comment (rghetia) [reply ocagent.go]: yes for panic. for the same reasons as for vmmetricreceiver [comment](https://github.com/open-telemetry/opentelemetry-service/pull/31#pullrequestreview-252602891)

File: options.go [hunk 3]
```diff
@@ -30,6 +31,22 @@ type ExporterOption interface {
...
+
+func (rd resourceDetector) withExporter(e *Exporter) {
+	e.resourceDetector = resource.Detector(rd)
+}
+
+// WithResource allows one to register a resource detector. Resource Detector is used
+// to detect resources associated with the application. Detected resource is exported
+// along with the metrics. If the detector fails then it panics.
+// If a resource detector is not provided then by default it detects from the environment.
+func WithResource(rd resource.Detector) ExporterOption {
```
Comment (songy23) [inline options.go]: Nit: `WithResourceDetector`

Comment (rghetia) [reply options.go]: done.

**Pull request merged**